### ISSUE TITLE
Thrown an error when a private method has MVC annotations

### ISSUE THREAD

**Issue by danielfireman** [jooby-project/jooby#372]
I've just ran into this problem. For instance:

```
public class PingRoute {
  @Path("/ping")
  @GET
   private ping() {
     return Results.ok()
   }
}
```

If I `use(PingRoute.class)`, the ping endpoint is silently ignored. Although I know that we could easily add unit tests to find this error, I think that we could also follow a fail-fast approach and prevent the server from start (giving a nice error message). 

What do you guys think? I am happy to give it a shot as soon I get some sort of thumbs up.

**Comment by jknack** [comment]
We used to have the check, but then I removed it [here](https://github.com/jooby-project/jooby/commit/69281b16b5ff55c756883ba1bb8e0fbfb8c81dca)

I removed bc it was dead code. For example the ```.getMethods()``` only returns public methods, before we were using ```.getDeclaredMethods```

The *challenge* here is to supports #365 in an easy way and taking care of ```method overrides```. This was the main reason of the change.

I'm cool to getting this back and there are enough tests to validate what I explained here.

Thanks!
